power-policy-service/src/consumer.rs

```diff
@@ -13,24 +14,30 @@ pub struct State {
     /// The ID of the currently connected consumer
     device_id: DeviceId,
     /// The power capability of the currently connected consumer
-    power_capability: PowerCapability,
+    consumer_power_capability: ConsumerPowerCapability,
 }
 
-impl PartialOrd for State {
-    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
-        Some(self.cmp(other))
-    }
-}
-
-impl Ord for State {
-    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
-        self.power_capability.cmp(&other.power_capability)
+/// Compare two consumer capabilities to determine which one is better
+///
+/// This is not part of the `Ord` implementation for `ConsumerPowerCapability`, because it's specific to this implementation
+fn cmp_consumer_capability(a: &ConsumerPowerCapability, b: &ConsumerPowerCapability) -> Ordering {
+    // Intially compare power first
+    let ord = a.capability.cmp(&b.capability);
+    if ord == Ordering::Equal {
+        // Prioritize unconstrained devices
+        match (a.flags.unconstrained_power(), b.flags.unconstrained_power()) {
+            (true, false) => Ordering::Greater,
+            (false, true) => Ordering::Less,
+            _ => Ordering::Equal,
+        }
+    } else {
+        ord
     }
```

Comment:
This `unconstrained_power` ordering is just the default `impl Ord for bool` therefore this is the default Ord for a tuple
```suggestion
    // Intially compare power first, but prioritize unconstrained devices if equal
    (a.capability, a.flags.unconstrained_power()).cmp(&(b.capability, b.flags.unconstrained_power()))
```

Reply:
Done.